opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2430,11 +2475,29 @@ namespace detail {
             pres = V::Zero(dims);
 
             for (int c = 0; c < nc; ++c) {
-                if (fipnum[c] != 0 && mask[c]) {
-                    hcpv[fipnum[c]-1] += pv[c] * hydrocarbon[c];
-                    pres[fipnum[c]-1] += pv[c] * pressure.value()[c];
-                    values[fipnum[c]-1][5] += pv[c];
-                    values[fipnum[c]-1][6] += pv[c] * pressure.value()[c] * hydrocarbon[c];
+                const int region = fipnum[c] - 1;
+                if (region != -1 && mask[c]) {
+                    hcpv[region] += pv[c] * hydrocarbon[c];
+                    pres[region] += pv[c] * pressure.value()[c];
+                }
+            }
+
+            sd_.fip[SimulatorData::FIP_PV] = V::Zero(nc);
+            sd_.fip[SimulatorData::FIP_WEIGHTED_PRESSURE] = V::Zero(nc);
+
+            for (int c = 0; c < nc; ++c) {
+                const int region = fipnum[c] - 1;
+                if (region != -1 && mask[c]) {
+                    sd_.fip[SimulatorData::FIP_PV][c] = pv[c];
+
+                    if (hcpv[region] != 0) {
+                        sd_.fip[SimulatorData::FIP_WEIGHTED_PRESSURE][c] = pv[c] * pressure.value()[c] * hydrocarbon[c] / hcpv[region];
```

Comment:
This is the cause for the differences that @GitPaean sees. hcpv is not correct if a region crosses into the domain of another process.That is why this division was moved after the global sum